frontend/public/components/default-resource.jsx

```diff
@@ -17,30 +18,65 @@ import {
 const { common } = Cog.factory;
 const menuActions = [...common];
 
-const Header = props => <ListHeader>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.name">Name</ColHead>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.namespace">Namespace</ColHead>
-  <ColHead {...props} className="col-sm-4 hidden-xs" sortField="metadata.creationTimestamp">Created</ColHead>
-</ListHeader>;
+const computeColumnSizes = (columns, isResourceNamespaced) => {
+  const columnsNumber = isResourceNamespaced ? _.size(columns) + 1 : _.size(columns);
+  const columnsSize = Math.floor(12 / columnsNumber);
+  const fistColumnSize = 12 - (columnsSize * (columnsNumber-1));
+  return {
+    fistColumnSize,
+    columnsSize
+  };
+};
+
+const CustomHeader_ = props => {
+  if (!_.get(props, ['resourceTable', 'table'])) {
+    return null;
+  }
+  const { table, isNamespaced } = props.resourceTable;
+  const { fistColumnSize, columnsSize } = computeColumnSizes(table.columnDefinitions, isNamespaced);
+  const columnHeaders = _.map(table.columnDefinitions, (column, index) => {
+    if (index === 0) {
+      return <ColHead key={index} className={`col-xs-${fistColumnSize}`}>{column.name}</ColHead>;
+    }
+    return <ColHead key={isNamespaced ? index + 1 : index} className={`col-xs-${columnsSize}`}>{column.name}</ColHead>;
+  });
+  isNamespaced && columnHeaders.splice(1, 0, (<ColHead key={1} className={`col-xs-${columnsSize}`}>Namespace</ColHead>));
+  return <ListHeader>
+    {columnHeaders}
+  </ListHeader>;
+};
 
-const RowForKind = kind => function RowForKind_ ({obj}) {
+const stateToProps = ({UI}) => {
+  const resourceTable = UI.get('resourceTable');
+  return {resourceTable};
+};
+
+const CustomHeader = connect(stateToProps)(CustomHeader_);
+
+const RowForKind = (kind, rows, isNamespaced) => function RowForKind_ ({obj, index}) {
+  if (!rows) {
+    return null;
+  }
+  const row = rows[index];
+  const { cells } = row;
+  const namespace = _.get(row, 'object.metadata.namespace');
+  const { fistColumnSize, columnsSize } = computeColumnSizes(cells, isNamespaced);
+  const renderedRows = _.map(cells, (column, i) => {
+    if (i === 0 ) {
+      return <div key={i} className={`col-xs-${fistColumnSize} co-resource-link-wrapper`}>
+        <ResourceCog actions={menuActions} kind={referenceFor(obj) || kind} resource={obj} />
+        <ResourceLink kind={kind} name={obj.metadata.name} namespace={obj.metadata.namespace} title={obj.metadata.name} />
+      </div>;
+    }
+    return <div key={isNamespaced ? i + 1 : i} className={`col-xs-${columnsSize} co-break-word`}>{column}</div>;
```

Comment:
We need to make sure we're formatting the column based on the type. Dates and possibly other types should be rendered specially instead of using the raw value that's returned.

Reply:
So in case of namespace we want to show the *resource-icon* infront and have the namespace as link ?

> Dates and possibly other types should be rendered specially instead of using the raw value that's returned.

So in case of **Age** we should use **Created** for header and have the output showing the same values as the ones rendered in the **Created** column (eg. Secrets table) ?
I guess other types will emerge during the work

Reply:
The API returns a `type` property for the data. We should be looking at that to know how to format it.